navigator_helpers/tasks/sample_to_dataset/task_suite.py

```diff
@@ -895,3 +901,95 @@ def process_row(
             generated_data_w_seeds_df = pd.DataFrame()
 
         return generated_data_df, generated_data_w_seeds_df
+
+    def preprocess_sample_dataset(
+        self,
+        sample_dataset: pd.DataFrame,
+        system_prompt_type: str = "cognition",
+        num_samples: int = 25,
+    ) -> pd.DataFrame:
```

Comment:
Putting these two tasks (column name normalization and clipping num samples) together works here, but we'll probably want to split them up when we move over to Jarvis - e.g., we'll likely want the check on num_records to happen client-side to limit the amount of data hitting that gets sent to our API.